dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -165,8 +169,12 @@ internal class DatadogRumMonitor(
 
     internal fun handleEvent(event: RumRawEvent) {
         handler.removeCallbacks(keepAliveRunnable)
-        synchronized(rootScope) { rootScope.handleEvent(event, writer) }
-        handler.postDelayed(keepAliveRunnable, KEEP_ALIVE_MS)
+        executorService.submit {
+            synchronized(rootScope) {
```

Comment:
hmmm....do you want to apply synchronisation only at the level of the DatadogRumMonitor instance ? Otherwise I would create a static locker object and synchronise around that this way you are safe even if there are hypothetically 2 monitors running in the same time.

Reply:
No I don't need any more synchronisation, first because there can't be 2 different monitors, and second because the synchronisation only concerns the rootScope field, which is local to the monitor instance anyway.